Dockerfile.freyja_apps.multi

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+# SPDX-License-Identifier: MIT
+
+# Comments are provided throughout this file to help you get started.
+# If you need more help, visit the Dockerfile reference guide at
+# https://docs.docker.com/engine/reference/builder/
+
+################################################################################
+# Create a stage for building the application.
+
+ARG RUST_VERSION=1.72.1
+ARG APP_NAME=freyja-in-memory-app
+ARG UID=10001
+
+# CONFIG_PATH must be relative to the project root
+ARG CONFIG_PATH=""
+
+FROM --platform=$BUILDPLATFORM docker.io/library/rust:${RUST_VERSION} AS build
+
+# Target architecture to cross-compile
+ARG TARGETARCH
+
+ARG APP_NAME
+ARG CONFIG_PATH
+WORKDIR /sdv
+
+COPY ./ .
+
+COPY ./container/cargo/config.toml ./.cargo/config.toml
+
+# Check that APP_NAME argument is valid.
+RUN /sdv/container/scripts/argument_sanitizer.sh \
+    --arg-value "${APP_NAME}" \
+    --regex "^[a-zA-Z_0-9-]+$" || \
+    ( echo "Argument sanitizer failed for ARG 'APP_NAME'"; exit 1 )
+
+# Check that TARGETARCH argument is valid.
+RUN /sdv/container/scripts/argument_sanitizer.sh \
+    --arg-value "${TARGETARCH}" \
+    --regex "^[a-zA-Z_0-9-]+$" || \
+    ( echo "Argument sanitizer failed for ARG 'TARGETARCH'"; exit 1 )
+
+# Check that CONFIG_PATH argument is valid if it's non-empty.
+RUN if [ -n "${CONFIG_PATH}" ]; then \
+        /sdv/container/scripts/argument_sanitizer.sh \
+        --arg-value "${CONFIG_PATH}" \
+        --regex "^[.\/a-zA-Z_0-9-]+$" || \
```

Comment:
The '.' typically represents any character.  I think (not 100% sure) that you may need to use the escape character '\' before the '.'.  Please check this.

Reply:
Most special characters don't need to be escaped in character sets. See https://www.regular-expressions.info/charclass.html#:~:text=Your%20regex%20will%20work%20fine,escape%20it%20with%20another%20backslash.

Reply:
Lauren brought up a decent point that even if its not needed now a change to the script may inadvertently cause issues. the . is escaped in the latest change but I am open to removing it again.

Reply:
Yeah, I agree that it is not necessary for this case, but I thought it was okay to escape it anyway for human readability/ in case someone were to ever change the way the r3egex is being interpreted in the future. But I am open to either way since both work